### PR TITLE
feature: move snapshot management to container's create and remove

### PR DESCRIPTION
**Pull request by skoo87**
Signed-off-by: skoo87 <marckywu@gmail.com>

**1.Describe what this PR did**
Now, when create a container, will create a snapshot with container's
id. After container be stopped, the snapshot will not be removed until
the container be removed.

**2.Does this pull request fix one issue?** 

**3.Describe how you did it**

**4.Describe how to verify it**

**5.Special notes for reviews**

### CODE REVIEW

**Comment by skoo87** [comment]
@allencloud please review it.

File: daemon/mgr/container_types.go [hunk 1]
```diff
@@ -64,6 +64,11 @@ func (c *Container) ID() string {
 	return c.meta.ID
 }
 
+// Image returns container's image name.
+func (c *Container) Image() string {
+	return c.meta.Config.Image
```
Comment (allencloud) [inline daemon/mgr/container_types.go]: Currently, In mgr's code, there are so many pointer calling. As a result, if struct has many pointers, there will be more potential panic there. I tested some and find some obvious panic, such as https://github.com/alibaba/pouch/issues/352.

Maybe we need to add more rule of contractual programming right now to guarantee the quality of code.

Comment (skoo87) [reply daemon/mgr/container_types.go]: Many pointers is not a problem, We don't need to be afraid of this. such as "c.meta.Config.Image", We must guarantee the "C" (Container) is valid at the origin position only, so the pointers don't be nil.

Comment (allencloud) [reply daemon/mgr/container_types.go]: > We must guarantee the "C" (Container) is valid at the origin position only

👍 

This conversation is not blocking this pull request.

**Comment by skoo87** [comment]
ping @allencloud

**Comment by allencloud** [comment]
LGTM

**Pull request merged**